lib/node_modules/@stdlib/_tools/remark/plugins/remark-lint-html-section-structure/lib/main.js

```diff
@@ -0,0 +1,366 @@
+* @param {Object} node - AST node to attach message to
+* @param {string} message - error message
+* @param {string} errorCode - error code identifier
+*/
+function reportErr( file, node, message, errorCode ) {
+	var fullMsg = formatPosition( node ) + '   error   ' + message + '   ' + errorCode;
+	debug( fullMsg );
+	file.message( fullMsg, node );
+}
+
+/**
+* Extracts comment class name from a node.
+*
+* @private
+* @param {Object} node - AST node containing comment
+* @param {RegExp} matchRegex - regular expression for matching comment
+* @param {RegExp} simpleMatchRegex - alternative regular expression for matching comment
+* @returns {string|null} extracted class name or null if no match
+*/
+function extractCommentClass( node, matchRegex, simpleMatchRegex ) {
+	var simpleMatch;
+	var match;
+
+	match = matchRegex.exec( node.value );
+	if ( match ) {
+		return match[ 1 ];
+	}
+
+	simpleMatch = simpleMatchRegex.exec( node.value );
+	if ( simpleMatch ) {
+		return trim( simpleMatch[ 1 ] );
+	}
+
+	return null;
+}
+
+/**
+* Validates that the class name in a comment matches the section class.
+*
+* @private
+* @param {Object} file - virtual file
+* @param {Object} node - node containing the comment
+* @param {string} cClass - class name extracted from comment
+* @param {string} sClass - class name from section
+* @param {string} errMsg - error message to use if validation fails
+* @param {string} errCode - error code to use if validation fails
+* @returns {boolean} boolean indicating whether the comment class matches the section class
+*/
+function checkCommentClass( file, node, cClass, sClass, errMsg, errCode ) {
+	if ( sClass && cClass !== sClass ) {
+		debug( 'Mismatched section class in comment: expected %s, got %s', sClass, cClass );
+		errMsg = replace( errMsg, '%s1', sClass );
+		errMsg = replace( errMsg, '%s2', cClass );
+		reportErr( file, node, errMsg, errCode );
+		return false;
+	}
+	return true;
+}
+
+/**
+* Validates if there is sufficient whitespace between nodes.
+*
+* @private
+* @param {Object} endNode - end section node
+* @param {Object} commentNode - comment node
+* @returns {boolean} boolean indicating whether there is sufficient whitespace between nodes
+*/
+function hasValidWhitespace( endNode, commentNode ) {
+	var lineGap = commentNode.position.start.line - endNode.position.end.line;
+	debug( 'Gap between section end and comment: %d lines', lineGap );
+
+	// We require at least 2 lines of gap for an empty line:
+	return lineGap >= 2;
+}
+
+/**
+* Checks if a text node contains proper empty line.
+*
+* @private
+* @param {Object} node - text node to check
+* @returns {boolean} boolean indicating whether the node contains proper empty line
+*/
+function hasEmptyLine( node ) {
+	var value;
+
+	if ( node.type !== 'text' ) {
+		return false;
+	}
+
+	value = node.value;
+
+	// Look for at least two consecutive newlines (representing an empty line):
+	return (/(\r?\n)\s*(\r?\n)/).test( value );
```

Comment:
Is there a reason not to move this regexp to the parent scope to be with the other regexp?

Reply:
Moved. Thanks for noticing!